packages/playwright-core/src/server/registry/oopDownloadBrowserMain.ts

```diff
@@ -104,11 +107,78 @@ function downloadFile(options: DownloadParams): Promise<void> {
   }
 }
 
+async function throwUnexpectedResponseError(response: http.IncomingMessage) {
+  let body = '';
+  try {
+    await new Promise<void>((resolve, reject) => {
+      response
+          .on('data', chunk => body += chunk)
+          .on('end', resolve)
+          .on('error', reject);
+    });
+  } catch (error) {
+    body += error;
+  }
+
+  response.resume(); // consume response data to free up memory
+
+  throw new Error(`server returned code ${response.statusCode} body '${body}'`);
+}
+
+async function downloadAndExtractBrotli(options: DownloadParams) {
+  const response = await new Promise<http.IncomingMessage>((resolve, reject) => httpRequest({
+    url: options.url,
+    headers: {
+      'User-Agent': options.userAgent,
+    },
+    timeout: options.connectionTimeout,
+  }, resolve, reject));
+
+  log(`-- response status code: ${response.statusCode}`);
+  if (response.statusCode !== 200)
+    await throwUnexpectedResponseError(response);
+
+  const totalBytes = parseInt(response.headers['content-length'] || '0', 10);
+  log(`-- total bytes: ${totalBytes}`);
+
+  let downloadedBytes = 0;
+  response.on('data', chunk => {
+    downloadedBytes += chunk.length;
+    progress(downloadedBytes, totalBytes);
+  });
+
+  await pipeline(
+      response,
+      createBrotliDecompress(),
+      tarFs.extract(options.browserDirectory, {
+        map(header: { name: string }) {
+          // manually-created ffmpeg archive has files nested in subdirectory
```

Comment:
lets try to fix that?

Reply:
the [new ffmpeg roll](https://github.com/microsoft/playwright/pull/34079) was built on CI and doesn't have this problem anymore.